client/dashboard/sites/index.tsx

```diff
@@ -146,3 +150,63 @@ export default function Sites() {
 		</>
 	);
 }
+
+function recordViewChanges(
+	oldView: View,
+	newView: View,
+	recordTracksEvent: ReturnType< typeof useAnalytics >[ 'recordTracksEvent' ]
+) {
+	if ( oldView.type !== newView.type ) {
+		recordTracksEvent( 'calypso_dashboard_sites_list_type_changed', { type: newView.type } );
+
+		// Changing view type can also change fields, but they weren't triggered by a user
+		// action, so we won't record those tracks events.
+		return;
+	}
+
+	if (
+		oldView.sort?.field !== newView.sort?.field ||
+		oldView.sort?.direction !== newView.sort?.direction
+	) {
+		recordTracksEvent( 'calypso_dashboard_sites_list_sort_changed', {
+			field: newView.sort?.field,
+			direction: newView.sort?.direction,
+		} );
+	}
+
+	const oldFilterFields = new Set( oldView.filters?.map( ( { field } ) => field ) || [] );
+	const newFilterFields = new Set( newView.filters?.map( ( { field } ) => field ) || [] );
+
+	// @ts-ignore Set.difference should be available in Hosting Dashboard environments
+	for ( const added of newFilterFields.difference( oldFilterFields ) ) {
+		recordTracksEvent( 'calypso_dashboard_sites_list_filter_changed', {
+			change: 'added',
+			field: added,
+		} );
+	}
+	// @ts-ignore Set.difference should be available in Hosting Dashboard environments
+	for ( const removed of oldFilterFields.difference( newFilterFields ) ) {
+		recordTracksEvent( 'calypso_dashboard_sites_list_filter_changed', {
+			change: 'removed',
+			field: removed,
+		} );
+	}
+
+	const oldShownFields = new Set( oldView.fields || [] );
+	const newShownFields = new Set( newView.fields || [] );
+
+	// @ts-ignore Set.difference should be available in Hosting Dashboard environments
+	for ( const added of newShownFields.difference( oldShownFields ) ) {
+		recordTracksEvent( 'calypso_dashboard_sites_list_field_visibility_changed', {
```

Comment:
(brainstorming) should we also add something like 

`calypso_dashboard_sites_list_field_final` (bad naming)

which lists the final fields? Basically we might want to know what the final set of fields that our users finally settle on.

Reply:
🤔 seems sort of complicated. Would it be a throttled function that assumed the field choices were final if no changes had been made in the last 30 seconds?

I suspect we're not going to be interested in what the most popular permutation of fields are. We would find that 5% of users use the same permutation, and that's the most popular, so we make it the default.

What would be better would be to find the top field that users turn on most often and turn that on by default e.g. we find that "visitors" is enabled by 80% of users who tweak their fields. That 80% turn on a single field feels like a much better signal than 5% of users who use the exact permutation of fields.

Reply:
>  seems sort of complicated. Would it be a throttled function that assumed the field choices were final if no changes had been made in the last 30 seconds?

You're right, without that logic we wouldn't know how to collect that data. 🤦 

> What would be better would be to find the top field that users turn on most often and turn that on by default e.g. we find that "visitors" is enabled by 80% of users who tweak their fields. That 80% turn on a single field feels like a much better signal than 5% of users who use the exact permutation of fields.

This logic is so much better and reliable 😄 Thanks!